### ISSUE TITLE
about TBROK causing test failure

### ISSUE THREAD

**Issue by qqxlt** [linux-test-project/ltp#974]
Some test cases fail. It can be seen from the test process that it is caused by TBROK. Repeated testing of the test case has a probability of passing. Can this count as failure of the test case?

for example
![memcg_failcnt_test1](https://user-images.githubusercontent.com/21022560/191909301-4a4f1543-a22b-482c-a355-11884fe66c6c.png)

**Comment by metan-ucw** [comment]
I/O errors from memcg files looks like a something is wrong with your kernel.

Also TBROK means that test setup has failed and something is broken, it's not the cause but rather end result of broken system.

**Comment by qqxlt** [comment]
> Repeated testing of the test case has a probability of passing.

Repeated testing of the test case has a probability of passing. Do you still need to pay attention to such mistakes?

**Comment by metan-ucw** [comment]
Of course, if test fails only once in ten or even in hundred runs there is likely problem somewhere. Linux kernel is massively parallel and most of the cases you need right timing for a race condition to happen. Sometimes the probability of hitting the bug is really small, but that does not mean that it's not there.

**Comment by qqxlt** [comment]
> Of course, if test fails only once in ten or even in hundred runs there is likely problem somewhere. Linux kernel is massively parallel and in most of the cases you need right timing for a race condition to happen. Sometimes the probability of hitting the bug is really small, but that does not mean that it's not there.

Thanks

**Comment by qqxlt** [comment]
> > Of course, if test fails only once in ten or even in hundred runs there is likely problem somewhere. Linux kernel is massively parallel and in most of the cases you need right timing for a race condition to happen. Sometimes the probability of hitting the bug is really small, but that does not mean that it's not there.
> 
> Thanks

Is there any good debugging method for such problems?

**Comment by metan-ucw** [comment]
I'm afraid there isn't easy manual to follow. Linux kernel is a complex beast and it takes time to understand its internals. I would have started with isolating the exact failure i.e. figuring out what step went wrong. Once you have idea what exactly went wrong you can start with debugging. However if this is a kernel race condition it takes years to acquire right skill set to be able to pinpoint the problem.

**Comment by qqxlt** [comment]
> I'm afraid there isn't easy manual to follow. Linux kernel is a complex beast and it takes time to understand its internals. I would have started with isolating the exact failure i.e. figuring out what step went wrong. Once you have idea what exactly went wrong you can start with debugging. However if this is a kernel race condition it takes years to acquire right skill set to be able to pinpoint the problem.

Thanks
